src/dpp/utility.cpp

```diff
@@ -299,7 +299,48 @@ namespace dpp {
 		uint32_t cmyk(int c, int m, int y, int k) {
 			return cmyk(c / 255.0, m / 255.0, y / 255.0, k / 255.0);
 		}
+
+		uint32_t hsl(int h, int s, int l) {
+			double hue = static_cast<double>(h) / 360.0;
+			double saturation = static_cast<double>(s) / 100.0;
+			double lightness = static_cast<double>(l) / 100.0;
+			return hsl(hue, saturation, lightness);
+		}
 		
+		uint32_t hsl(double h, double s, double l) {
+			const auto hue_to_rgb = [](double p, double q, double t){
+				if (t < 0) {
+					t += 1;
+				}
+				if (t > 1) {
```

Comment:
I feel this should be an else if, so that we're not double checking if the top value was already done (watch someone snipe this saying that doing `if() if()` is better than `if() else if()`. It could also just be a ternary operator.